lib/socket.js

```diff
@@ -66,10 +67,19 @@ class Socket extends EventEmitter {
     assert(socket.remoteAddress);
     assert(socket.remotePort != null);
     assert(ws);
+    assert(typeof req.url === 'string');
 
     let proto = 'ws';
     let host = socket.remoteAddress;
     let port = socket.remotePort;
+    const qs = '/?transport=websocket';
+    const qsIndex = req.url.indexOf(qs);
+
+    const namespace = req.url.slice(qsIndex);
+    assert(
+      namespace.indexOf('/?') !== -1,
```

Comment:
Should this be `namespace.indexOf('/?') === -1`?

Reply:
Yes. Thanks. Fixed!